x/participation/keeper/msg_participate.go

```diff
@@ -31,5 +56,21 @@ func (k msgServer) Participate(goCtx context.Context, msg *types.MsgParticipate)
 		return nil, sdkerrors.Wrap(types.ErrInvalidBidder, err.Error())
 	}
 
+	// set used allocations
+	usedAllocations, _ := k.GetUsedAllocations(ctx, msg.Participant)
```

Comment:
Does it returns an empty `usedAllocations` when not found?

I would prefer to explicitly initialize the `usedAllocations` structure if `GetUsedAllocations` returns `found==false` even if it makes the code a bit more complex, we are supposed to ignore `usedAllocations` value if `found==false`

Reply:
Yeah it returns an empty struct and since we are setting both fields, the effect is the same as initializing the empty struct.  I'll refactor to make it more readable though